util/stmtsummary/statement_summary.go

```diff
@@ -578,19 +579,25 @@ func (ssbd *stmtSummaryByDigest) toCurrentDatum(beginTimeForCurInterval int64) [
 
 	// `ssElement` is lazy expired, so expired elements could also be read.
 	// `beginTime` won't change since `ssElement` is created, so locking is not needed here.
-	if ssElement == nil || ssElement.beginTime < beginTimeForCurInterval {
+	if user == nil {
+		return ssElement.toDatum(ssbd)
+	}
+	if ssElement == nil || ssElement.beginTime < beginTimeForCurInterval || user == nil ||
+		(strings.Compare(ssElement.sampleUser, user.Username) != 0 && !isSuper) {
```

Comment:
`sampleUser` is one of all the users that requested this type of SQL.
You can check the place `sampleUser` is assigned.